src/v/wasm/logger.h

```diff
@@ -16,4 +16,34 @@
 namespace wasm {
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 extern ss::logger wasm_log;
+
+class logger {
```

Comment:
Thoughts on moving this to `api.h`? Since this is a public interface we expect integrating subsystems to implement, I think this makes since. As it stands it is a little awkward to have this in same file as the internal logger for the wasm subsystem.

Reply:
done